api/v3/OSF/Contract.php

```diff
@@ -186,178 +79,50 @@ function _civicrm_api3_o_s_f_contract_process(&$params) {
-        'financial_type_id' => 2, // Membership Dues
-        'payment_instrument_id' => $payment_instrument,
-      ]);
-      // reload mandate
-      $mandate = civicrm_api3('SepaMandate', 'getsingle', [
-        'check_permissions' => 0,
-        'id' => $mandate['id']
-      ]);
-      $bank_account = _civicrm_api3_o_s_f_contract_getBA($params['iban'], $params['contact_id']);
-      // create the contract
-      $result = civicrm_api3('Contract', 'create', [
-        'check_permissions' => 0,
-        'sequential' => empty($params['sequential']) ? 0 : 1,
-        'contact_id' => $params['contact_id'],
-        'membership_type_id' => $params['membership_type_id'],
-        'join_date' => $params['member_since'],
-        'start_date' => $params['start_date'],
-        'source' => 'OSF',
-        'campaign_id' => $params['campaign_id'],
-        'membership_payment.membership_annual' => number_format($params['amount'] * $params['frequency'], 2, '.', ''),
-        'membership_payment.membership_frequency' => $params['frequency'],
-        'membership_payment.membership_recurring_contribution' => $mandate['entity_id'],
-        'membership_payment.payment_instrument' => $payment_instrument,
-        'membership_payment.to_ba' => _civicrm_api3_o_s_f_contract_getBA($creditor['iban'], GPAPI_GP_ORG_CONTACT_ID, []),
-        'membership_payment.from_ba' => $bank_account,
-        'membership_payment.cycle_day' => $cycle_day,
-      ]);
+      $contract_helper = \Civi\Gpapi\ContractHelper\Factory::createWithoutExistingMembership($params);
+      $contract_id = $contract_helper->create($params);
     } catch (Exception $ex) {
       throw $ex;
     } finally {
       $lock->release();
     }
 
-    $activity_id = civicrm_api3('Activity', 'getvalue', [
-      'return' => 'id',
-      'activity_type_id' => 'Contract_Signed',
-      'source_record_id' => $result['id'],
-    ]);
-
-    CRM_Gpapi_Processor::updateActivityWithUTM($params, $activity_id);
+    $null = NULL;
 
-    // get BA reference type for IBAN to do BankingAccountReference lookup by type
-    $reference_type_iban = civicrm_api3('OptionValue', 'getvalue', [
-      'return' => 'id',
-      'option_group_id' => 'civicrm_banking.reference_types',
-      'value' => 'IBAN',
-    ]);
-    $bank_account_reference = civicrm_api3('BankingAccountReference', 'getvalue', [
-      'return' => 'id',
-      'ba_id' => $bank_account,
-      'reference_type_id' => $reference_type_iban,
-    ]);
-    $reference_type_id = civicrm_api3('OptionValue', 'getvalue', [
-      'return' => 'id',
-      'option_group_id' => 'civicrm_banking.reference_types',
-      'value' => $referenceType,
-    ]);
-    // update the bank account reference type (CE always creates IBAN)
-    civicrm_api3('BankingAccountReference', 'create', [
-      'reference_type_id' => $reference_type_id,
-      'id' => $bank_account_reference,
-    ]);
+    return civicrm_api3_create_success(
+      [['id' => $contract_id]],
```

Comment:
`$contract_id` is actually an array with all the contract data rather than just the ID.